File: app/schemas/backup.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class BackupScenarioCreate(BaseModel):
    career_id: UUID
    scenario_name: str = Field(..., max_length=500)
    description: str | None = None
    transition_difficulty: str | None = None
    estimated_transition_months: int | None = None
    reasoning: str | None = None


class BackupPlanCreate(BaseModel):
    title: str | None = None
    description: str | None = None
    scenarios: list[BackupScenarioCreate]


class BackupScenarioResponse(BaseModel):
    id: UUID
    career_id: UUID
    scenario_name: str
    description: str | None = None
    transition_difficulty: str | None = None
    estimated_transition_months: int | None = None
    reasoning: str | None = None

    model_config = {"from_attributes": True}


class BackupPlanResponse(BaseModel):
    id: UUID
    user_id: UUID
    title: str | None = None
    description: str | None = None
    status: str
    scenarios: list[BackupScenarioResponse] = []
    created_at: datetime

    model_config = {"from_attributes": True}


class BackupPlanListResponse(BaseModel):
    items: list[BackupPlanResponse]
    total: int
    page: int
    page_size: int
    total_pages: int

File: src/types/backup.ts
import type { TransitionDifficulty } from "@/lib/constants";

export interface BackupPlan {
  id: string;
  user_id: string;
  title: string | null;
  description: string | null;
  status: string;
  created_at: string;
  scenarios: BackupScenario[];
}

export interface BackupScenario {
  id: string;
  backup_plan_id: string;
  career_id: string;
  scenario_name: string;
  description: string | null;
  transition_difficulty: TransitionDifficulty | null;
  estimated_transition_months: number | null;
  reasoning: string | null;
  career?: {
    id: string;
    title: string;
    description: string;
    average_salary: number | null;
  };
}

export interface BackupPlanListResponse {
  items: BackupPlan[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface BackupScenarioCreatePayload {
  career_id: string;
  scenario_name: string;
  description?: string;
  transition_difficulty?: TransitionDifficulty;
  estimated_transition_months?: number;
  reasoning?: string;
}

export interface BackupPlanCreatePayload {
  title?: string;
  description?: string;
  scenarios: BackupScenarioCreatePayload[];
}
